File: asyncawait-ts/basics02.ts
import {sleep} from "./lib";

function main() {
    (async () => {
        const p = doWorld();
        console.log('Hello');
        await p;
    })();
    // cannot block
}

async function doWorld() {
    await sleep(1000);
    console.log('World!');
}

main();

File: goroutine/basics02.go
package main

import (
	"sync"
	"time"
)

func main() {
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		defer wg.Done()
		doWorld02()
	}()
	println("Hello")
	wg.Wait()
}

func doWorld02() {
	time.Sleep(time.Second)
	println("World!")
}
